source/python/idetect/nlp_models/relevance/relevance.py

```diff
@@ -0,0 +1,205 @@
+import os
+import requests
+import gensim
+import pandas as pd
+import numpy as np
+from nltk.tokenize import WordPunctTokenizer
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.pipeline import Pipeline
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.externals import joblib
+
+from idetect.model import Relevance
+
+class RelevanceModel(object):
+    def __init__(self, model_path=None):
+        self.model = self.load_model(model_path=model_path)
+
+    def load_model(self, model_path=None):
+        if model_path and os.path.isfile(model_path):
+            clf = joblib.load(model_path)
+        else:
+            default_model_path = 'relevance_classifier.pkl'
+            if os.path.isfile(default_model_path):
+                clf = joblib.load(default_model_path)
+            else:
+                url = 'https://s3-us-west-2.amazonaws.com/idmc-idetect/relevance_models/relevance_classifier.pkl'
+                r = requests.get(url, stream=True)
+                if not os.path.isfile(default_model_path):
+                    try:
+                        os.makedirs(os.path.dirname(default_model_path))
+                    except OSError as exc: # Guard against race condition
+                        if exc.errno != errno.EEXIST:
+                            raise
+                with open(default_model_path, 'wb') as f:
+                    for chunk in r.iter_content(chunk_size=1024):
+                        if chunk:  # filter out keep-alive new chunks
+                            f.write(chunk)
+                clf = joblib.load(default_model_path)
+        return clf
+
+    def predict(self, text):
+        try:
+            relevance = self.model.transform(pd.Series(text))[0]
+        except:
+            # if error occurs, classify as most likely category
+            relevance = 'yes'
+
+        if relevance == 'yes':
+            return Relevance.DISPLACEMENT
+        else:
+            return Relevance.NOT_DISPLACEMENT
+
+class Tokenizer(TransformerMixin):
```

Comment:
Generally, I think we should repeat as little code as possible. If we find a bug in Tokenizer, we don't want to have to make sure to fix it in both places.